### ISSUE TITLE
HelperProvider ve dvou extensionach: "Service '24.helperProvider.0' has already been added."

### ISSUE THREAD

**Issue by jirinapravnik** [flame-org/Modules#25]
Pokud použiju konfiguraci helperů, dle tohoto rázu:
https://gist.github.com/jsifalda/7f570f94974b62163117

tak když mám ve dvou extension. Padne mi to na tomto:  Service '24.helperProvider.0' has already been added.

Problém je na tomto řádku: https://github.com/flame-org/Modules/blob/master/Flame/Modules/DI/ModulesExtension.php#L191 - protože to přebírá $key z pole...

**Comment by jsifalda** [comment]
Co toto https://github.com/flame-org/Modules/commit/e0cc52661fe5e75f7a1bf160a55306a4ec034304 & https://github.com/flame-org/Modules/commit/ac9dcb75a0e33caf32c6e79ec3cf27404aea203c ? jsem otevren alternativnim moznostem. :)

**Comment by jirinapravnik** [comment]
Ted jsem na to mrkl, a co úplně jednoduše udělat ten řádek:

$provider = $builder->addDefinition($this->prefix('helperProvider.' . str_replace('\\', '', $helper)))

vezme to plny nazev souboru a nahradí jen zpetna lomitka z namespaces, ktera byt nesmi v nazvu?

**Comment by jsifalda** [comment]
to ale porad nevyresi problem, pokud tam pridas ten sami helper dvakrat?

**Comment by jirinapravnik** [comment]
To ne, ale to ja ani neresil. Me dela problem tohle:
ArticleModuleExtension - tam mam ArticleHelperModule - pridam v extension (ma to jakoby key, tedy i name 0)
PhotoModuleExtension - tam mam PhotoHelperModule - pridam v extension (ma to taky jakoby key, tedy i name 0) a tak se to bije

To je podle me mnohem zavaznejsi prusvih, ale mohli bychom vyresit rovnou i to, jak rikas ty, asi to taky obcas muze nastat

**Comment by jsifalda** [comment]
tak to prijde moje reseni jako lepsi, protoze resi oboje situace? (snad) :blush:

**Comment by jirinapravnik** [comment]
Odzkousel jsem, a zda se to ok, tak snad to bude ok:-) Btw neplanujes vydat novy stable? Ja ted zjistil, ze je vubec nejaky novy zpusob konfigurace a dle me celkem povedeny:-) I když bylo by hezčí zbavit se toho ampersandu v argumentu...

**Comment by jsifalda** [comment]
No, to je neco co me docela tizi... a nova konfigurace neni uplne dokoncena. Pokud si dobre pamatuji, mel jsem tam problem s konfiguraci routes. 
Coz prinasi i takove neprijemnosti, jakoze nemuzu vydat setinkovou verzi, ktera by opravovala tvuj problem... (nebo alespon nevim jak)

**Comment by jirinapravnik** [comment]
Nejsem příliš znalý GITu, ale není od tohohle cherry-pick? vytáhne commit a soupne ho do vetve? Nicmene ja t uopravu nepotrebuji zatim, upravil jsem si to pirmo ve vendoru docasne.

Nicmene zkusil jsem dev-master - v projektu co je ve vyvoji a zadny problem s novou konfiguraci jsem zatim nenarazil... routy pridavam pres routerfactory services

**Comment by jsifalda** [comment]
V tomto pripade nam bohuzel cherry-pick nepomuze... pac opravny commit pracuje se souborem [TemplateHelpersConfig.php](https://github.com/flame-org/Modules/blob/e0cc52661fe5e75f7a1bf160a55306a4ec034304/Flame/Modules/Configurators/TemplateHelpersConfig.php) ten vsak v poslednim stable vubec neni... 
Dalsi problem je to, ze bohuzel nemam vubec cas na to, udelat update http://flame-org.github.io/Modules/ pro novou konfiguraci.
Nemel by si zajem @jirinapravnik ? :)

**Comment by jirinapravnik** [comment]
Kdyby to bylo v češtině tak klidně, ale moje angličtina je spíše na úrovni nutného čtení, psaní by dobře nedopadlo:/

i když je taky otazkou, zda bych byl ten spravny, protože ne vše používám, nevím, jestli má být dále funkční starý způsob apod.

**Comment by jsifalda** [comment]
Tak to by se dalo nejak vymyslet :)
Na tu anglictinu muzu kouknout, i kdyz to neni ani moje silna stranka...
